### ISSUE TITLE
NPE when calling "getValues(true)" on a BigDecimalRange without a step (step = null).

### ISSUE THREAD

**Issue by thake** [faktorips/faktorips.base#6]
A NPE occurs when calling "size()" on a BigDecimalRange without a step (step = null).

The method "sizeForDiscreteValuesExcludingNull" should IMHO look like this:
```java
    @Override
    protected int sizeForDiscreteValuesExcludingNull() {
        if(getStep() == null){
            return Object.equals(this.getLowerBound(), this.getUpperBound()) ? 1 : Integer.MAX_VALUE;
        }
        BigDecimal size = getUpperBound().subtract(getLowerBound()).abs()
                .divide(getStep(), 0, RoundingMode.UNNECESSARY).add(BigDecimal.ONE);
        if (size.longValue() > Integer.MAX_VALUE) {
            throw new RuntimeException(
                    "The number of values contained within this range is to huge to be supported by this operation.");
        }
        return size.intValue();
    }
```

**Comment by hd42** [comment]
This has already been fixed in FIPS-7316 / [305f407](https://github.com/faktorips/faktorips.base/commit/305f4070dbe3ed334060ddddca29afab9beedd14) for Faktor-IPS 21.6. sizeForDiscreteValuesExcludingNull() will only be called if step!=null

**Comment by hd42** [comment]
When calling size(), there should not be an NPE here, because sizeForDiscreteValuesExcludingNull() is only called if the range isDiscrete(), which is false if the step is null

**Comment by thake** [comment]
@hd42 you are right. I size() is actually not the problem. The NPE I'm experiencing is caused by "getValues(true)" on a BigDecimalRange without a step.

Here is the test:
```java
@Test
void testGetValuesOfOfBigDecimalRangeWithoutStep(){
  var range = new BigDecimalRange(valueOf(10), valueOf(10));
  assertThat(range.getValues(true)).contains(valueOf(10));
}
```

Another test that I would expect to pass but that currently fails is the following:
```java
@Test
void testSizeOfBigDecimalRangeWithEqualBoundariesHavingDifferentPrecision(){
  var range = new BigDecimalRange(new BigDecimal("10"), new BigDecimal("10.00"));
  assertThat(range.size()).isEqualTo(1);
}
```

**Comment by hd42** [comment]
Thank you for those tests. Is there a specific Faktor-IPS version you require a fix for?

**Comment by thake** [comment]
You're welcome :) Thanks for the good software!

Next upcoming Faktor-IPS version is fine for me. No need to backport.

**Comment by hd42** [comment]
Fixed in the upcoming release 21.12
